File: family_order/html/person/addgroupmemberbyorder/scripts/AddGroupmemberByOrder.js
//集团资料查询成功后调用的方法
function selectGroupAfterAction(data) {
	var custName = data.get("CUST_NAME");
	var groupId = data.get("GROUP_ID");
	var custId = data.get("CUST_ID");
	$("#GROUP_ID_NAME").val(custName);//界面将集团客户编码转换成名称
	$("#GROUP_CUST_ID").val(groupId);//集团客户编码
	$("#GROUP_CUST_NAME").val(custName);
	$("#GROUP_CUSTID").val(custId);//集团的custId(集团客户标识)

}
//集团资料查询失败后调用的方法
function selectGroupErrorAfterAction() {
	alert("查询数据出错");
	return;
}
//批量导入
function batImportBook(){
//	backPopup('UI-popup');
	var importFile = $("#FILE_ID").val();
    if (importFile == "") {
        alert("请选择需要导入的文件！");
        return;
    }
	$.beginPageLoading("正在录入数据...");
	var param = "&params=" + importFile;
	ajaxSubmit(null, "batImportBook",
			param, null, function(data) {
				$.endPageLoading();
				if (data.get('SUCCESS') == 'SUCCESS') {    //弹出返回的页面提示信息
				   alert("导入数据成功");
				}else{
					alert("导入数据出错");
				}
				
			}, function(error_code, error_info, derror) {
				$.endPageLoading();
				alert(error_info);
			});
}
function qryButton(){
	var serial = $("#SERIAL_NUMBER1").val();
	if (serial == '') {
		alert("请输入手机号码");
		return;
	}
	ajaxSubmit('searchPart','querySubscriber','','submitPart',function(data){
		var alertInfo = data.get('ALERT_INFO');
		var remark898 = data.get('REMARK898');
		var groupbook = data.get('GROUPBOOK');
		var remarkvpmn = data.get('REMARKVPMN');
		
		$("#ALERT_INFO").val(alertInfo);
		if (alertInfo == '1') {    //弹出返回的页面提示信息
			$("#SERIAL_NUMBER1").val(serial);
			alert("非正常用户，不能成为集团通讯录成员！");
			return;
		}
		//移动用户
		var mobileFlag = $("#IS_MOBILE").val();
		if(mobileFlag == '1'){
			$("#GROUP_898").attr('disabled',true);
			$("#CITY_CODE").attr('disabled',true);
			$("#USEPSPT_TYPE_CODE").attr('disabled',true);
			$("#GROUP_BOOK_MEMBER").attr('disabled',true);
			$("#GROUNP_V_MEMBER").attr('disabled',true);
			$("#USEPSPT_END_DATE").attr('disabled',true);
			
			$("#CUST_NAME").attr('disabled',true);
			$("#USECUST_NAME").attr('disabled',true);
			$("#SERIAL_NUMBER").attr('disabled',true);
			$("#EPARCHY_NAME").attr('disabled',true);
			$("#USEPSPT_ID").attr('disabled',true);
			$("#USEPSPT_ADDR").attr('disabled',true);
			$("#USEPSPT_END_DATE").attr('disabled',true);
			$("#EPARCHY_CODE").attr('disabled',true);
		}
		if(remark898 == '2'){
			$("#GROUNP_NAME").attr('disabled',true);
		}
		if(groupbook == '2'){
			$("#GROUP_ID").attr('disabled',true);
			$("#GROUP_CUST_NAME1").attr('disabled',true);
			$("#IS_REMOVE").attr('disabled',true);
			
		}
		if(remarkvpmn == '2'){
			$("#VPN_NAME").attr('disabled',true);
		}
		}, function(code, info, detail) {
		    $.endPageLoading();
		    MessageBox.error("错误提示", info, function(btn) {
		    }, null, detail);
		   }, function() {
		    $.endPageLoading();
		    MessageBox.alert("告警提示", "操作超时!", function(btn) {
		    });
		  });

}
function setMessageStatus(){
	var memberKind = $("#MEMBER_KIND").val();
	if(memberKind == '2'){
		document.getElementById("CLASS_CONTACT").classList.add("e_required");
	}
	if(memberKind == '0' || memberKind == '1'){
		document.getElementById("CLASS_CONTACT").classList.remove("e_required");
	}
	
}
function queryStatus(){
	var hideFlag = $("#SELECT_NUM").val();
	if(hideFlag == '2'){//根据客户姓名录入
		$("#CUST_NAME").val("");
		$("#USECUST_NAME").val("");
		$("#SERIAL_NUMBER").val("");
		$("#MEMBER_KIND").val("");
		$("#IS_CONTACT").val("");
		$("#IS_MOBILE").val("");
		$("#EPARCHY_CODE").val("");
		$("#CITY_CODE").val("");
		$("#USEPSPT_TYPE_CODE").val("");
		$("#USEPSPT_ID").val("");
		$("#USEPSPT_ADDR").val("");
		$("#USEPSPT_END_DATE").val("");
		$("#USEPOST_ADDR").val("");
		$("#USEPHONE").val("");
		$("#SVC_MODE_CODE").val("");
		$("#DEPART").val("");
		$("#DUTY").val("");
		$("#GROUP_898").val("");
		$("#GROUNP_NAME").val("");
		$("#GROUNP_V_MEMBER").val("");
		$("#VPN_NAME").val("");
		$("#GROUP_BOOK_MEMBER").val("");
		$("#GROUP_ID").val("");
		$("#GROUP_CUST_NAME1").val("");
		$("#IS_REMOVE").val("");
		$("#LIKE_MOBILE_BUSI").val("");
		$("#LIKE_DISCNT_TYPE").val("");
		$("#LIKE_ACT").val("");
		$("#REMARK").val("");
		
		
		
		$("#bquery").css("display","none");
		$("#NO_PHONE").val("1");
		$("#GROUP_898").attr('disabled','');
		$("#CITY_CODE").attr('disabled','');
		$("#USEPSPT_TYPE_CODE").attr('disabled','');
		$("#GROUP_BOOK_MEMBER").attr('disabled','');
		$("#GROUNP_V_MEMBER").attr('disabled','');
		$("#USEPSPT_END_DATE").attr('disabled','');
		
		$("#CUST_NAME").attr('disabled','');
		$("#USECUST_NAME").attr('disabled','');
		$("#SERIAL_NUMBER").attr('disabled','');
		$("#EPARCHY_NAME").attr('disabled','');
		$("#USEPSPT_ID").attr('disabled','');
		$("#USEPSPT_ADDR").attr('disabled','');
		$("#USEPSPT_END_DATE").attr('disabled','');
		$("#EPARCHY_CODE").attr('disabled','');
	}
	if(hideFlag == '1'){//根据服务号码查
		$("#bquery").css("display","");
		$("#NO_PHONE").val("2");
		
		$("#CUST_NAME").val("");
		$("#USECUST_NAME").val("");
		$("#SERIAL_NUMBER").val("");
		$("#MEMBER_KIND").val("");
		$("#IS_CONTACT").val("");
		$("#IS_MOBILE").val("");
		$("#EPARCHY_CODE").val("");
		$("#CITY_CODE").val("");
		$("#USEPSPT_TYPE_CODE").val("");
		$("#USEPSPT_ID").val("");
		$("#USEPSPT_ADDR").val("");
		$("#USEPSPT_END_DATE").val("");
		$("#USEPOST_ADDR").val("");
		$("#USEPHONE").val("");
		$("#SVC_MODE_CODE").val("");
		$("#DEPART").val("");
		$("#DUTY").val("");
		$("#GROUP_898").val("");
		$("#GROUNP_NAME").val("");
		$("#GROUNP_V_MEMBER").val("");
		$("#VPN_NAME").val("");
		$("#GROUP_BOOK_MEMBER").val("");
		$("#GROUP_ID").val("");
		$("#GROUP_CUST_NAME1").val("");
		$("#IS_REMOVE").val("");
		$("#LIKE_MOBILE_BUSI").val("");
		$("#LIKE_DISCNT_TYPE").val("");
		$("#LIKE_ACT").val("");
		$("#REMARK").val("");
	}
}
function submitData(){
	var selectNum = $("#SELECT_NUM").val();//2根据号码查   1根据姓名查
	var queryFlag = $("#QUERY_FLAG").val();//1是查询过了
	if('1' == selectNum && '1' != queryFlag){
		alert("根据号码录入，请先点击查询客户资料！");
		return;
	}
	var serial = $("#SERIAL_NUMBER1").val();
	if (serial == '') {
		alert("服务号码或客户姓名为空，请先输入号码查询客户资料或输入客户姓名再录入");
		return;
	}
	var serialPh = $("#SERIAL_NUMBER").val();
	if (serialPh == '') {
		alert("电话号码不可为空！");
		return;
	}
	var custNameEx = $("#CUST_NAME").val();
	if (custNameEx == '') {
		alert("开户客户不可为空！");
		return;
	}
	var usercustNameEx = $("#USECUST_NAME").val();
	if (usercustNameEx == '') {
		alert("使用客户不可为空！");
		return;
	}
	var eparchyCodeEx = $("#EPARCHY_CODE").val();
	if (eparchyCodeEx == '') {
		alert("归属地州不可为空！");
		return;
	}
	var cityCodeEx = $("#CITY_CODE").val();
	if (cityCodeEx == '') {
		alert("归属业务区不可为空！");
		return;
	}
	var memberId = $("#MEMBER_ID").val();
	var groupCustName = $('#MEMBER_GROUP_CUST_NAME').val();
	var alertInfo = $("#ALERT_INFO").val();
	var groupCustId = $("#GROUP_CUST_ID").val();
	var isRemove = $("#IS_REMOVE").val();
	if(alertInfo == '1'){
		alert("非正常用户，不能成为集团通讯录成员！");
		return;
	}
	var memberKind = $("#MEMBER_KIND").val();
	if(memberKind == ''){
		alert("成员类型不可为空！");
		return;
	}
	var memberKind = $("#MEMBER_KIND").val();
	var isContact = $("#IS_CONTACT").val();
	if(memberKind == '2'){
		if(isContact == ''){
			alert("是否可接触服务客户不可为空！");
			return;
		}
	}
	
	if (memberId != '') {
			if(isRemove == ''){
				alert("请选择是否迁移");
				return;
			}
			if(isRemove == '1'){
				if(groupCustId == ''){
					alert("集团客户名称不能为空");
					return;
				}
				var memberGroupId = $("#MEMBER_GROUP_ID").val();
				var groupCustId = $("#GROUP_CUST_ID").val();
				var CustName = $("#GROUP_CUST_NAME").val();
				
				if(memberGroupId == groupCustId){
					alert("该号码已是:"+CustName+"集团成员，不允许重复迁移同一集团！");
					return;
				}
			}
		    if(isRemove == '2'){
			    alert("该号码已在集团通讯录，不允许提交！");
				  return;
		    }
		MessageBox.confirm("告警提示",'该号码已是'+groupCustName+'集团的通讯录成员，是否继续?',function(ret){
			  if(ret=="ok"){
			   ajaxSubmit('submitPart','submitCheck','','',function(data){
				   if (data.get('SUCCESS') == 'SUCCESS') {    //弹出返回的页面提示信息
					   alert("录入成功");
					   $("#QUERY_FLAG").val('2');
					}else{
						alert("录入数据出错");
						
						$("#QUERY_FLAG").val('2');
					}
					return;
					}, function(code, info, detail) {
					    $.endPageLoading();
					    MessageBox.error("错误提示", info, function(btn) {
					    }, null, detail);
					   }, function() {
					    $.endPageLoading();
					    MessageBox.alert("告警提示", "操作超时!", function(btn) {
					    });
					  });
			  }
		});
	}else{
		if(groupCustId == ''){
			alert("集团客户名称不能为空");
			return;
		}
		MessageBox.confirm("提示",'确认提交?',function(ret){
			  if(ret=="ok"){
				  ajaxSubmit('submitPart','submitCheck','','',function(data){
					   if (data.get('SUCCESS') == 'SUCCESS') {    //弹出返回的页面提示信息
						   alert("录入成功");
						   $("#QUERY_FLAG").val('2');
						}else{
							alert("录入数据出错");
							$("#QUERY_FLAG").val('2');
						}
						return;
						}, function(code, info, detail) {
						    $.endPageLoading();
						    MessageBox.error("错误提示", info, function(btn) {
						    }, null, detail);
						   }, function() {
						    $.endPageLoading();
						    MessageBox.alert("告警提示", "操作超时!", function(btn) {
						    });
						  });
			  }
		});
		
	}
	
	

}
/**
 *导入模板 
 */
function openHref(){
	window.location.href='attach?action=downloadweb&realName=TXLMEMBER.xlsx&filePath=template/bat/group/TXLMEMBER.xlsx'
}

File: family_order/src/web/com/asiainfo/veris/crm/order/web/person/addgroupmemberbyorder/AddGroupmemberByOrder.java
package com.asiainfo.veris.crm.order.web.person.addgroupmemberbyorder;
import org.apache.tapestry.IRequestCycle;

import com.ailk.common.data.IData;
import com.ailk.common.data.impl.DataMap;
import com.ailk.common.util.parser.ExcelConfig;
import com.ailk.common.util.parser.ImpExpUtil;
import com.asiainfo.veris.crm.order.pub.exception.BatException;
import com.asiainfo.veris.crm.order.pub.frame.bcf.util.IDataUtil;
import com.asiainfo.veris.crm.order.pub.frame.bcf.util.SysDateMgr;
import com.asiainfo.veris.crm.order.web.frame.bcf.base.CSViewCall;
import com.asiainfo.veris.crm.order.web.frame.bcf.exception.CSViewException;
import com.asiainfo.veris.crm.order.web.frame.csview.person.base.PersonBasePage;
public abstract class AddGroupmemberByOrder extends PersonBasePage {
	

	public abstract void setMember(IData member);
    public abstract void setInfo(IData info);
	public abstract void setCond(IData cond);
	public abstract void setVpmn(IData vpmn);
	public abstract void setSeral(IData seral);
	public abstract void setQryflag(IData qryflag);
	
	public void querySubscriber(IRequestCycle cycle)throws Exception{
		
		IData qryFlag = new DataMap();
		qryFlag.put("QUERY_FLAG", "1");
		setQryflag(qryFlag);//主要用于判断是否查询过
		String alertInfo = "";
		String remark898 = "";
		String groupBook = "";
		String remarkVpmn = "";
		IData param = getData();
		IData inParam = new DataMap();
		IData serialData = new DataMap();
		serialData.put("SERIAL_NUMBER1", param.getString("SERIAL_NUMBER1"));
		this.setSeral(serialData);
		inParam.put("SERIAL_NUMBER1", param.getString("SERIAL_NUMBER1"));
		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.querySubscriber", inParam);
		if(IDataUtil.isNotEmpty(result)){
			String mobileFlag = result.getString("MOBILE_FLAG");
			if(mobileFlag == null || mobileFlag == ""){
				//移动号码
				alertInfo = "0";
	        	String userId = result.getString("USER_ID");
	        	//判断号码是否属于898集团成员
	        	remark898 = queryGroup(userId);
	        	//判断是否通讯录成员
	        	groupBook = queryMember(userId);
	        	//判断号码是否属于集团V网成员
	        	remarkVpmn = queryGroupVpmn(userId);
	        	IData iData = new DataMap();
	    		iData.put("ALERT_INFO", alertInfo);
	    		iData.put("REMARK898", remark898);
	    		iData.put("GROUPBOOK", groupBook);
	    		iData.put("REMARKVPMN", remarkVpmn);
	    		setAjax(iData);
	    		result.put("MOBILE_FLAG", "1");//移动号码
	    		this.setInfo(result);
			}else{
				//异网号码
				//判断是否通讯录成员
	        	groupBook = queryMember(param.getString("SERIAL_NUMBER1"));//异网号码的userid为手机号码
	        	IData iData = new DataMap();
	        	iData.put("GROUPBOOK", groupBook);
	        	setAjax(iData);
				this.setInfo(result);
			}
		}else{
			alertInfo = "1";
			IData iData = new DataMap();
			iData.put("ALERT_INFO", alertInfo);
			iData.put("REMARK898", remark898);
			iData.put("GROUPBOOK", groupBook);
			iData.put("REMARKVPMN", remarkVpmn);
			setAjax(iData);
			this.setInfo(result);
		}
	}
	/**
	 * 判断号码是否属于898集团成员
	 * @param cycle
	 * @throws Exception
	 */
	public String queryGroup(String userId)throws Exception{
		String remark898 = "";
		IData inParam = new DataMap();
		inParam.put("USER_ID", userId);
		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.queryGroup", inParam);
		if (IDataUtil.isEmpty(result))
        {
			result = new DataMap();
			remark898 = "2";
        }else{
        	remark898 = "1";
        }
		result.put("REMARK898", remark898);
		this.setCond(result);
		return remark898;
		
	}
	/**
	 * 系统判断号码是否重复加集团通讯录成员
	 * @param userId
	 * @throws Exception
	 */
	public String queryMember(String userId)throws Exception{
		String groupBook = "";
		IData inParam = new DataMap();
		inParam.put("USER_ID", userId);
		
		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.queryMember", inParam);
		if (IDataUtil.isEmpty(result))
        {
			result = new DataMap();
			groupBook = "2";
        }else{
        	groupBook = "1";
        }
		result.put("GROUPBOOK", groupBook);
		this.setMember(result);
		return groupBook;
		
	}
	/**
	 * 判断号码是否属于集团V网成员
	 * @param userId
	 * @throws Exception
	 */
	public String queryGroupVpmn(String userId)throws Exception{
		String remarkVpmn = "";
		IData inParam = new DataMap();
		inParam.put("USER_ID", userId);
		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.queryGroupVpmn", inParam);
		
		if (IDataUtil.isEmpty(result))
        {
			result = new DataMap();
			remarkVpmn = "2";
        }else{
        	remarkVpmn = "1";
        }
		result.put("REMARKVPMN", remarkVpmn);
		this.setVpmn(result);
		return remarkVpmn;
		
	}
	/**
	 * 批量导入通讯录数据数据
	 * @param userId
	 * @throws Exception
	 */
	public void batImportBook(IRequestCycle cycle)throws Exception{
		
		IData data = getData();
        String fileId = data.getString("params");
        ImpExpUtil.getImpExpManager().getFileAction().setVisit(getVisit());
        IData fileData = ImpExpUtil.beginImport(null, fileId, ExcelConfig.getSheets("import/bat/group/BOOKMEMBER.xml"));
        int rightCount = fileData.getInt("rightCount", 0);
        int errorCount = fileData.getInt("errorCount", 0);

        if (IDataUtil.isEmpty(fileData)) {
            CSViewException.apperr(BatException.CRM_BAT_89);
        }
        if (rightCount + errorCount == 0) {
            CSViewException.apperr(BatException.CRM_BAT_86);
        }
        if (rightCount > 0) {
            data.put("FILEDATA", fileData);
            IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.bookMemberImportOrder", data);
            String rtnResult = result.getString("SUCCESS");
    		this.setAjax("SUCCESS", rtnResult);// 传给页面提示
        } else {
            CSViewException.apperr(BatException.CRM_BAT_6);
        }
        
    }
		
		

	/**
	 * 提交录入数据
	 * @param userId
	 * @throws Exception
	 */
	public void submitCheck(IRequestCycle cycle)throws Exception{
		IData iparam = getData();
		//判断是否异网号码，异网号码录入数据封装，IS_MOBILE=2为异网
		String noPhone = iparam.getString("NO_PHONE");//判断根据客户姓名还是手机号码查
		if("1".equals(noPhone)){//根据客户姓名录入
			IData param = new DataMap();
			param.put("SERIAL_NUMBER", iparam.getString("SERIAL_NUMBER1"));
			param.put("CUST_ID", iparam.getString("GROUP_CUSTID"));//集团的custid
			param.put("PARTITION_ID", "1111");
			param.put("USER_ID", "1111");
			param.put("GROUP_ID", iparam.getString("GROUP_CUST_ID"));
			param.put("MEMBER_CUST_ID", "1111");//用户的custid
			param.put("USECUST_ID", "1111");
			param.put("CUST_NAME", iparam.getString("CUST_NAME"));//开户客户
			param.put("USECUST_NAME", iparam.getString("USECUST_NAME"));//使用客户
			param.put("EPARCHY_CODE", iparam.getString("EPARCHY_CODE"));
    		param.put("CITY_CODE", iparam.getString("CITY_CODE"));
    		param.put("JOIN_DATE", SysDateMgr.getSysTime());
    		param.put("REMOVE_TAG", "0");
    		param.put("MEMBER_KIND", iparam.getString("MEMBER_KIND"));
    		param.put("IS_CONTACT", iparam.getString("IS_CONTACT"));
    		param.put("GROUP_CUST_NAME", iparam.getString("GROUP_CUST_NAME"));
    		param.put("USEPSPT_TYPE_CODE", iparam.getString("USEPSPT_TYPE_CODE"));
    		param.put("USEPSPT_ID", iparam.getString("USEPSPT_ID"));
    		param.put("USEPSPT_ADDR", iparam.getString("USEPSPT_ADDR"));
    		param.put("USEPSPT_END_DATE", iparam.getString("USEPSPT_END_DATE"));
    		param.put("USEPOST_ADDR", iparam.getString("USEPOST_ADDR"));
    		param.put("USEPHONE", iparam.getString("USEPHONE"));
    		param.put("SVC_MODE_CODE", iparam.getString("SVC_MODE_CODE"));
    		param.put("DEPART", iparam.getString("DEPART"));
    		param.put("DUTY", iparam.getString("DUTY"));
    		param.put("LIKE_MOBILE_BUSI", iparam.getString("LIKE_MOBILE_BUSI"));
    		param.put("LIKE_DISCNT_TYPE", iparam.getString("LIKE_DISCNT_TYPE"));
    		param.put("LIKE_ACT", iparam.getString("LIKE_ACT"));
    		param.put("REMARK", iparam.getString("REMARK"));
    		param.put("RSRV_TAG5", "1");
    		
    		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.submitCheck", param);
    		String rtnResult = result.getString("SUCCESS");
    		this.setAjax("SUCCESS", rtnResult);// 传给页面提示
    		
		}else{
			String isMobile = iparam.getString("IS_MOBILE");
			if("2".equals(isMobile)){
				String isRemove = iparam.getString("IS_REMOVE");
				if("1".equals(isRemove)){
					//更新状态
					String userId = iparam.getString("SERIAL_NUMBER1");//异网userid为手机号码
					IData inparams = new DataMap();
		    		inparams.put("USER_ID", userId);
		    		CSViewCall.call(this, "SS.ReqGroupmemberByOrderSVC.submitCancel", inparams);
				}
				//异网号码不做信息查询判断直接录入
				IData param = new DataMap();
				param.put("IS_MOBILE", iparam.getString("IS_MOBILE"));
				param.put("PARTITION_ID", Long.parseLong(iparam.getString("SERIAL_NUMBER1"))%10000);
	    		param.put("USER_ID", iparam.getString("SERIAL_NUMBER1"));
	    		param.put("CUST_ID", iparam.getString("GROUP_CUSTID"));//集团的custid
	    		param.put("GROUP_ID", iparam.getString("GROUP_CUST_ID"));
	    		param.put("MEMBER_CUST_ID", iparam.getString("SERIAL_NUMBER1"));//用户的custid
	    		param.put("USECUST_ID", iparam.getString("SERIAL_NUMBER1"));
	    		param.put("SERIAL_NUMBER", iparam.getString("SERIAL_NUMBER1"));
	    		param.put("CUST_NAME", iparam.getString("CUST_NAME"));//开户客户
	    		param.put("USECUST_NAME", iparam.getString("USECUST_NAME"));//使用客户
	    		param.put("EPARCHY_CODE", iparam.getString("EPARCHY_CODE"));
	    		param.put("CITY_CODE", iparam.getString("CITY_CODE"));
	    		param.put("JOIN_DATE", SysDateMgr.getSysTime());
	    		param.put("REMOVE_TAG", "0");
	    		param.put("MEMBER_KIND", iparam.getString("MEMBER_KIND"));
	    		param.put("IS_CONTACT", iparam.getString("IS_CONTACT"));
	    		
	    		param.put("GROUP_CUST_NAME", iparam.getString("GROUP_CUST_NAME"));
	    		param.put("USEPSPT_TYPE_CODE", iparam.getString("USEPSPT_TYPE_CODE"));
	    		param.put("USEPSPT_ID", iparam.getString("USEPSPT_ID"));
	    		param.put("USEPSPT_ADDR", iparam.getString("USEPSPT_ADDR"));
	    		param.put("USEPSPT_END_DATE", iparam.getString("USEPSPT_END_DATE"));
	    		param.put("USEPOST_ADDR", iparam.getString("USEPOST_ADDR"));
	    		param.put("USEPHONE", iparam.getString("USEPHONE"));
	    		param.put("SVC_MODE_CODE", iparam.getString("SVC_MODE_CODE"));
	    		param.put("DEPART", iparam.getString("DEPART"));
	    		param.put("DUTY", iparam.getString("DUTY"));
	    		
	    		param.put("RSRV_TAG1", iparam.getString("GROUP_898"));
	    		param.put("RSRV_STR1", iparam.getString("GROUNP_NAME"));
	    		param.put("RSRV_TAG4", iparam.getString("GROUNP_V_MEMBER"));
	    		param.put("RSRV_STR4", iparam.getString("VPN_NAME"));
	    		param.put("RSRV_TAG5", "1");
	    		param.put("OLD_GROUP_ID", iparam.getString("GROUP_ID"));
	    		param.put("OLD_GROUP_CUST_NAME", iparam.getString("GROUP_CUST_NAME1"));
	    		param.put("RSRV_TAG2", iparam.getString("IS_REMOVE"));
	    		param.put("LIKE_MOBILE_BUSI", iparam.getString("LIKE_MOBILE_BUSI"));
	    		param.put("LIKE_DISCNT_TYPE", iparam.getString("LIKE_DISCNT_TYPE"));
	    		param.put("LIKE_ACT", iparam.getString("LIKE_ACT"));
	    		param.put("REMARK", iparam.getString("REMARK"));
	    		param.put("CUST_MANAGER_ID", iparam.getString("CUST_MANAGER_ID"));
	    		String fileId = iparam.getString("simpleUpload");
	    		param.put("RSRV_STR3", fileId);
	    		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.submitCheck", param);
	    		String rtnResult = result.getString("SUCCESS");
	    		this.setAjax("SUCCESS", rtnResult);// 传给页面提示
			}else{//移动号码
				//判断是否迁移，如果是迁移IS_REMOVE=1,原来用户tf_f_addressbook_member的remove_tag改为1
				//并新增一条记录tf_f_addressbook_member,记录原来的group_id group_cust_name 到
				//old_group_id old_group_cust_name
				String isRemove = iparam.getString("IS_REMOVE");
				if("1".equals(isRemove)){
					//更新状态
					String userId = iparam.getString("USER_ID");
					IData inparams = new DataMap();
		    		inparams.put("USER_ID", userId);
		    		CSViewCall.call(this, "SS.ReqGroupmemberByOrderSVC.submitCancel", inparams);
				}
	    		//新增记录
	    		String fileId = iparam.getString("simpleUpload");
	    		IData param = new DataMap();
	    		String user_id = iparam.getString("USER_ID");
	    		param.put("IS_MOBILE", iparam.getString("IS_MOBILE"));
	    		param.put("PARTITION_ID", Long.parseLong(user_id)%10000);
	    		param.put("USER_ID", user_id);
	    		param.put("CUST_ID", iparam.getString("GROUP_CUSTID"));//集团的custid
	    		param.put("GROUP_ID", iparam.getString("GROUP_CUST_ID"));
	    		param.put("MEMBER_CUST_ID", iparam.getString("CUST_ID"));//用户的custid
	    		param.put("USECUST_ID", iparam.getString("CUST_ID"));
	    		param.put("SERIAL_NUMBER", iparam.getString("SERIAL_NUMBER1"));
	    		param.put("CUST_NAME", iparam.getString("CUST_NAME"));
	    		param.put("USECUST_NAME", iparam.getString("CUST_NAME"));
	    		param.put("EPARCHY_CODE", iparam.getString("EPARCHY_CODE"));
	    		param.put("CITY_CODE", iparam.getString("CITY_CODE"));
	    		param.put("JOIN_DATE", SysDateMgr.getSysTime());
	    		param.put("REMOVE_TAG", "0");
	    		param.put("MEMBER_KIND", iparam.getString("MEMBER_KIND"));
	    		param.put("IS_CONTACT", iparam.getString("IS_CONTACT"));
	    		
	    		param.put("GROUP_CUST_NAME", iparam.getString("GROUP_CUST_NAME"));
	    		param.put("USEPSPT_TYPE_CODE", iparam.getString("USEPSPT_TYPE_CODE"));
	    		param.put("USEPSPT_ID", iparam.getString("USEPSPT_ID"));
	    		param.put("USEPSPT_ADDR", iparam.getString("USEPSPT_ADDR"));
	    		param.put("USEPSPT_END_DATE", iparam.getString("USEPSPT_END_DATE"));
	    		param.put("USEPOST_ADDR", iparam.getString("USEPOST_ADDR"));
	    		param.put("USEPHONE", iparam.getString("USEPHONE"));
	    		param.put("SVC_MODE_CODE", iparam.getString("SVC_MODE_CODE"));
	    		param.put("DEPART", iparam.getString("DEPART"));
	    		param.put("DUTY", iparam.getString("DUTY"));
	    		param.put("RSRV_TAG1", iparam.getString("GROUP_898"));
	    		param.put("RSRV_STR1", iparam.getString("GROUNP_NAME"));
	    		param.put("RSRV_TAG4", iparam.getString("GROUNP_V_MEMBER"));
	    		param.put("RSRV_STR4", iparam.getString("VPN_NAME"));
	    		param.put("RSRV_TAG5", "1");
	    		param.put("OLD_GROUP_ID", iparam.getString("GROUP_ID"));
	    		param.put("OLD_GROUP_CUST_NAME", iparam.getString("GROUP_CUST_NAME1"));
	    		param.put("RSRV_TAG2", iparam.getString("IS_REMOVE"));
	    		param.put("LIKE_MOBILE_BUSI", iparam.getString("LIKE_MOBILE_BUSI"));
	    		param.put("LIKE_DISCNT_TYPE", iparam.getString("LIKE_DISCNT_TYPE"));
	    		param.put("LIKE_ACT", iparam.getString("LIKE_ACT"));
	    		param.put("REMARK", iparam.getString("REMARK"));
	    		param.put("CUST_MANAGER_ID", iparam.getString("CUST_MANAGER_ID"));
	    		param.put("RSRV_STR3", fileId);
	    		IData result = CSViewCall.callone(this, "SS.AddGroupmemberByOrderSVC.submitCheck", param);
	    		String rtnResult = result.getString("SUCCESS");
	    		this.setAjax("SUCCESS", rtnResult);// 传给页面提示
				
				
			}
		}
		
		
	}
	
}
